xxh3.h

```diff
@@ -295,7 +295,7 @@ XXH3_len_1to3_64b(const void* data, size_t len, const void* keyPtr, XXH64_hash_t
         BYTE const c3 = ((const BYTE*)data)[len - 1];
         U32  const l1 = (U32)(c1) + ((U32)(c2) << 8);
         U32  const l2 = (U32)(len) + ((U32)(c3) << 2);
-        U64  const ll11 = XXH_mult32to64((l1 + seed + key32[0]), (l2 + key32[1]));
+        U64  const ll11 = XXH_mult32to64((l1 + seed + key32[0]), (l2 + key32[1])); // fixme: MSVC warns that seed is truncated to 32bits
```

Comment:
We should change XXH_mult32to64 to this then:
```c

/* U64 XXH_mult32to64(U32 a, U32 b) { return (U64)a * (U64)b; } */
#ifdef _MSC_VER
#   include <intrin.h>
    /* MSVC doesn't do a good job with the mull detection. */
#   define XXH_mult32to64(x, y) __emulu((U32)(x), (U32)(y))
#else
#   define XXH_mult32to64(x, y) ((U64)((x) & 0xFFFFFFFF) * (U64)((y) & 0xFFFFFFFF))
#endif
```

Reply:
OK, one issue is that the code source is under change currently.
Latest version is actually in `xxh3` branch.
This specific issue is fixed there.
There are likely other changes which would be of interest for this patch.

I suggest :
- either base your patch on `xxh3`, and PR into `cyan4973/xxh3`
- or wait for `xxh3` to be merged into `dev`. This will hopefully happen later in the week. Might be very soon if I break the changes into 2 parts (`_64bits` first, then `_128bits` later).

Reply:
If the truncation of the upper 32 bits is desired, then yes

Reply:
It's more than truncation, the new algorithm in the `xxh3` branch is different in non-trivial ways.

Reply:
Ok, I can wait for the merge and then either propagate it to my branch or fork it again from the most recent version and create new request.

Reply:
Any update on the right timing to fork the repo and fix the few glitches?